.github/workflows/build-push.yml

```diff
@@ -0,0 +1,110 @@
+
+name: Build and Push Docker Image WEAC Codebase
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'python-flask/**'
+      - 'c-sharp-razor/**'
+      - 'rust-actix/**'
+      - 'ruby-sinatra/**'
+      - 'javascript-express/**'
+      - 'go-gin/**'
+      - 'go-gorilla/**'
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        service:
+          - python-flask
+          - c-sharp-razor
+          - rust-actix
+          - ruby-sinatra
+          - javascript-express
+          - go-gin
+          - go-gorilla
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: '1'
+
+      - name: Get Changed Files
+        id: changed-files
+        uses: tj-actions/changed-files@v34
```

Comment:
I think for your case this dependency is good since you only want to track those services. 

The CI flow, however, needs to test all services excluding the workflow folder OR only certain services if there were no changes in root which speaks for a custom script.